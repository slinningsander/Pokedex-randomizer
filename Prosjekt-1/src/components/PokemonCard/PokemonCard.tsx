import { useNavigate } from 'react-router-dom';
import HeartComponent from '../HeartComponent/HeartComponent';
import './PokemonCard.css';

type PokemonCardProps = {
  name: string;
  type: string;
  imgURL: string;
  setRefresh: (refresh: boolean) => void;
  refresh: boolean;
};

export function PokemonCard({ name, type, imgURL, setRefresh, refresh }: PokemonCardProps) {
  const imgAltText = name + ' sprite';

  const pokemonSprite = <img className="pokemonSprite" src={imgURL} alt={imgAltText} />;

  const typePil = <span className="typePil">{type}</span>;

  const nameTitle = <h1 className="nameTitle">{name}</h1>;
  const navigate = useNavigate();
  const navigateToDetails = () => {
    navigate('/details/' + name);
  };

  const detailsButton = (
    <button className="detailsButton" onClick={navigateToDetails} data-testid={'test-detailsBtn'}>
      Details
    </button>
  );
  const heart = (
    <div className="heartContainer">
      <HeartComponent
        name={name}
        //Passes the setRefresh and refresh states to HeartComponent.
        setRefresh={setRefresh}
        refresh={refresh}
      />
    </div>
  );

  return (
    <div className="cardContainer">
      {pokemonSprite}
      {typePil}
      {nameTitle}
      {detailsButton}
      {heart}
    </div>
  );
}
export default PokemonCard;
