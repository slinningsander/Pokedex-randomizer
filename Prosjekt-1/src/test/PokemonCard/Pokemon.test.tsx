import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import PokemonCard from '../../components/PokemonCard/PokemonCard';
import catadoze from '/src/assets/catadoze.jpg';

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...(actual as object), // Wrap actual inside an object
    useNavigate: () => vi.fn(), // Mock useNavigate
  };
});

describe('PokemonCard-test', () => {
  beforeEach(() => {
    // mock localstorage
    const mockLocalStorage = {
      getItem: vi.fn().mockReturnValue('[]'),
      setItem: vi.fn(),
      clear: vi.fn(),
    };

    Object.defineProperty(window, 'localStorage', {
      value: mockLocalStorage,
    });
  });
  let mockRefresh = false;
  const mockSetRefresh = vi.fn((newRefresh) => {
    mockRefresh = newRefresh;
  });
  const pokemonCard = (
    <MemoryRouter>
      {' '}
      {/* Wrap your component in MemoryRouter */}
      <PokemonCard name="Pikachu" type="Electric" imgURL={catadoze} setRefresh={mockSetRefresh} refresh={mockRefresh} />
    </MemoryRouter>
  );

  it('Should find props', () => {
    render(pokemonCard);
    expect(screen.getByText('Pikachu')).toBeInTheDocument();
    expect(screen.getByText('Electric')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src', '/src/assets/catadoze.jpg');
  });

  it('has focus (button)', () => {
    render(pokemonCard);
    const button = screen.getByTestId('test-detailsBtn');
    button.focus();
    expect(button).toHaveFocus();
  });

  it('PokemonCard snapshot', () => {
    const { asFragment } = render(pokemonCard);
    expect(asFragment()).toMatchSnapshot();
  });
});
