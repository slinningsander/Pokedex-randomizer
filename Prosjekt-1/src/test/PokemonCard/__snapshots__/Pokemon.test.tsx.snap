// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard-test > PokemonCard snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="cardContainer"
  >
    <img
      alt="Pikachu sprite"
      class="pokemonSprite"
      src="/src/assets/catadoze.jpg"
    />
    <span
      class="typePil"
    >
      Electric
    </span>
    <h1
      class="nameTitle"
    >
      Pikachu
    </h1>
    <button
      class="detailsButton"
      data-testid="test-detailsBtn"
    >
      Details
    </button>
    <div
      class="heartContainer"
    >
      <svg
        class="notFav"
        data-testid="heartTestID"
        viewBox="0 0 32 29.6"
      >
        <filter
          id="inset-shadow"
        >
          <feoffset
            dx="0"
            dy="0"
          />
          <fegaussianblur
            result="offset-blur"
            stdDeviation="1"
          />
          <fecomposite
            in="SourceGraphic"
            in2="offset-blur"
            operator="out"
            result="inverse"
          />
          <feflood
            flood-color="black"
            flood-opacity=".95"
            result="color"
          />
          <fecomposite
            in="color"
            in2="inverse"
            operator="in"
            result="shadow"
          />
          <fecomposite
            in="shadow"
            in2="SourceGraphic"
            operator="over"
          />
        </filter>
        <path
          d="M23.6,0c-3.4,0-6.3,2.7-7.6,5.6C14.7,2.7,
    11.8,0,8.4,0C3.8,0,0,3.8,0,8.4c0,9.4,9.5,11.9,16,
    21.2c6.1-9.3,16-12.1,16-21.2C32,3.8,28.2,0,23.6,0z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
